common/util.cc

```diff
@@ -214,9 +214,9 @@ std::string hexdump(const uint8_t* in, const size_t size) {
 }
 
 std::string random_string(std::string::size_type length) {
-  const char* chrs = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  const std::string chrs {"0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"};
   std::mt19937 rg{std::random_device{}()};
-  std::uniform_int_distribution<std::string::size_type> pick(0, sizeof(chrs) - 2);
+  std::uniform_int_distribution<std::string::size_type> pick(0, chrs.length());
```

Comment:
```suggestion
  std::uniform_int_distribution<std::string::size_type> pick(0, chrs.length() - 1);
```

you're accessing out of bounds memory

Reply:
Good pickup! Confused by ASAN didn't throw an error on that when I ran it multiple times.

Reply:
Apparently str[str.size()] is valid in C++11, so presumably that isn't an out-of-bounds read to ASAN.

Super.